torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -983,6 +994,21 @@ def _supports_insertion(self, module: nn.Module) -> bool:
         # check if the module has any children and isn't observer
         num_children = len(list(module.children()))
         return num_children == 0 and not is_activation_post_process(module)
+        # check to see if module is of a supported type
+        num_children = len(list(module.children()))
+        is_supported_type = num_children == 0 and not isinstance(module, ObserverBase)
```

Comment:
@vspenubarthi Some merge-related issues? The code lines 997-999 is after `return` and will never be executed.

Reply:
@kit1980 seems like it, i'm surprised neither the linter nor tests found this. I created a new PR with the quick fix here: https://github.com/pytorch/pytorch/pull/81499

It would be great if you could give it a quick approval if you think it's all good.

Thank you for finding this.